src/test/java/com/google/gcp_variant_transforms/task/VcfToBqTaskTest.java

```diff
@@ -0,0 +1,112 @@
+import com.google.gcp_variant_transforms.options.VcfToBqContext;
+import com.google.gcp_variant_transforms.options.VcfToBqOptions;
+import com.google.inject.Injector;
+import java.io.IOException;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Mock;
+
+/**
+ * Units tests for VcfToBqTask.java
+ */
+ public class VcfToBqTaskTest {
+  Injector spyInjector;
+  VcfToBqTask vcfToBqTask;
+  VcfToBqTask spyVcfToBqTask;
+
+  @Mock
+  PipelineRunner mockedPipelineRunner;
+
+  @Mock
+  VcfToBqOptions mockedVcfToBqOptions;
+
+  @Mock 
+  VcfToBqContext mockedVcfToBqContext;
+
+  @Mock
+  HeaderReader mockedHeaderReader;
+
+  @Captor
+  ArgumentCaptor<PipelineOptions> argCaptorPipelineOptions;
+
+  @Captor
+  ArgumentCaptor<ImmutableList<String>> argCaptorImmutableList;
+
+  @Before
+  public void initializeSpyAndMocks() throws IOException{
+    MockitoAnnotations.initMocks(this);
+    vcfToBqTask = new VcfToBqTask(
+        mockedPipelineRunner, 
+        mockedHeaderReader, 
+        mockedVcfToBqContext, 
+        mockedVcfToBqOptions);
+    spyVcfToBqTask = spy(vcfToBqTask);
+  }
+
+  public void setUpRunBehaviors() throws IOException {
+    ImmutableList.Builder<String> headerLinesBuilder = new ImmutableList.Builder<String>();
+    for (int i = 0; i < 5; i++) {
+      headerLinesBuilder.add("##sampleHeaderLine=" + i);
+    }
+    ImmutableList<String> headerLines = headerLinesBuilder.build();
+    doReturn(headerLines).when(mockedHeaderReader).getHeaderLines();
+    doNothing().when(spyVcfToBqTask).setPipelineOptions(argCaptorPipelineOptions.capture());
+    doNothing().when(mockedVcfToBqContext).setHeaderLines(argCaptorImmutableList.capture());
+    doNothing().when(mockedPipelineRunner).runPipeline();
+  }
+
+  @Test
+  public void testVcfTaskConstructor_whenInitialize_thenNotNull() throws IOException {
+    assertThat(vcfToBqTask).isNotNull();
+    assertThat(spyVcfToBqTask).isNotNull();
+  }
+
+  @Test
+  public void testVcfTask_whenRun_thenVerifySetPipelineOptions() throws IOException {
+    setUpRunBehaviors();
+    spyVcfToBqTask.run();
+
+    verify(spyVcfToBqTask).setPipelineOptions(mockedVcfToBqOptions);    
+  }
+
+  @Test
+  public void testVcfTask_whenRun_thenVerifySetHeaderLines() throws IOException {
+    setUpRunBehaviors();
+    spyVcfToBqTask.run();
+
+    verify(mockedVcfToBqContext).setHeaderLines(argCaptorImmutableList.capture());  
+  }
+
+  @Test
+  public void testVcfTask_whenRun_thenVerifyRunPipeline() throws IOException {
+    setUpRunBehaviors();
+    spyVcfToBqTask.run();
+
+    verify(mockedPipelineRunner).runPipeline();  
+  }
+
+  @Test
+  public void testVcfTask_whenSetPipelineOptions_thenVerify() throws Exception {
+    doNothing().when(spyVcfToBqTask).setPipelineOptions(argCaptorPipelineOptions.capture());;
```

Comment:
This test is kinda pointless, tbh. We just call the function here and verify that function has been called. I say we remove it.